src/node/txdownloadman_impl.cpp

```diff
@@ -401,17 +410,8 @@ node::RejectedTxTodo TxDownloadManagerImpl::MempoolRejectedTx(const CTransaction
                 add_extra_compact_tx &= !m_orphanage.HaveTx(wtxid);
 
                 auto add_orphan_reso_candidate = [&](const CTransactionRef& orphan_tx, const std::vector<Txid>& unique_parents, NodeId nodeid, std::chrono::microseconds now) {
-                    const auto& wtxid = orphan_tx->GetWitnessHash();
-                    if (auto delay{OrphanResolutionCandidate(nodeid, wtxid, unique_parents.size())}) {
-                        const auto& info = m_peer_info.at(nodeid).m_connection_info;
+                    if (AddOrphanParentAnnouncements(unique_parents, orphan_tx->GetWitnessHash(), nodeid, now)) {
```

Comment:
I find the flow of logic a bit convoluted:

The check whether a peer is a orphan resolution candidate happens in `OrphanResolutionCandidate`, which also calculates the delay and is called from `AddOrphanParentAnnouncements`.

I would find the folllowing kind of of flow easier to understand (no need for `std::optional`, functions do exactly what their name suggests they do):
```
if (IsOrphanCandidate()) { // only checks if peer is a candidate
   AddOrphanParentAnnouncements(); // delay is calculated here or in a submethod
   m_orphanage.AddTx() / m_orphanage.AddAnnouncer();
}
```

Reply:
You're totally right that it's convoluted! I've changed the commit to just move all the parent inv logic to `OrphanResolutionCandidate` instead. We don't need 2 small helper functions where 1 always calls the other. I think this is even clearer, but let me know if you think I should still split it into `IsOrphanCandidate()` and `AddOrphanParentAnnouncements()`?

Reply:
Looks good to me now, rest is just a matter of taste.